features/dnsresolver/netty/src/main/java/org/opennms/netmgt/dnsresolver/netty/NettyResolverContext.java

```diff
@@ -145,25 +147,53 @@ public void destroy() {
     public CompletableFuture<Optional<String>> reverseLookup(InetAddress inetAddress) {
         final CompletableFuture<Optional<String>> future = new CompletableFuture<>();
         final String name = ReverseMap.fromAddress(inetAddress).toString();
+
+        // Netty does not perform caching when we query directly for DNS questions i.e. PTR requests
+        // so we perform the caching logic ourselves
+        final List<? extends DnsCacheEntry> entries = cache.get(name, null);
+        if (entries != null) {
+            // We've got a hit, so we have some cached result for this entry
+            // Try and find a hostname, or return an empty optional if none was found
+            final Optional<String> cachedHostname = entries.stream()
+                    .filter(e -> e instanceof ExtendedDnsCacheEntry)
+                    .map(e -> ((ExtendedDnsCacheEntry)e).hostnameFromPtrRecord())
+                    .filter(Objects::nonNull)
+                    .findFirst();
+            if (cachedHostname.isPresent()) {
+                // We found a cached hostname
+                return CompletableFuture.completedFuture(Optional.of(removeTrailingDot(cachedHostname.get())));
```

Comment:
We remove the trailing dot here, just to add it back later on.

Reply:
Where do we add it back on?

Reply:
Ok, now I get it. We ensure it's there first, then we remove it. Seems fine.